docs/madr/decisions/070-resource-identifier.md

```diff
@@ -0,0 +1,363 @@
+// StatsD only allow [a-zA-Z_\-0-9.] characters, everything else is removed
+// Extra dots breaks many regexes that converts statsd metric to prometheus one with tags
+func SanitizeMetric(metric string) string {
+	return illegalChars.ReplaceAllString(metric, "_")
+}
+```
+
+I can't find explanation why `alt_stat_name` and `stat_prefix` are treated differently by Envoy. 
+When Envoy exports metrics in Prometheus format resource name is always emitted as label value, it's never part of the metric name, i.e:
+```
+envoy_cluster_upstream_cx_total{envoy_cluster_name="localhost_5000"} 0
+```
+so there is no reason to replace `:` with `_`.
+
+If Envoy tried to support StatsD format then there are still a question, why `stat_prefix` emits metrics with `:` but `alt_stat_name` replaces `:` with `_`.
+Also, StatsD format doesn't seem to be supported out-of-the-box by Envoy and requires configuring `stats_sinks` in the static config.
+
+### Constraints
+
+#### What characters we can use in the identifier?
+ 
+In URL path segment we can use:
+
+```
+ALPHA / DIGIT / "-" / "." / "_" / "~" / "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "=" / ":" / "@"
+```
+
+In Envoy resource names we can use any character except `:` if we want to solve [Issue #2363](https://github.com/kumahq/kuma/issues/2363) without workaround.
+
+In Envoy stats fields we can use any character except `:` to avoid on the fly conversion of `:` to `_` for Cluster names in Prometheus labels.
+
+OpenTelemetry defines "Attribute", it's a key-value pair similar to Prometheus labels.
+There are [no charset limitation on attribute's key or value](https://opentelemetry.io/docs/specs/otel/common/?utm_source=chatgpt.com#attribute).
+
+This leaves us with the following charset:
+
+```
+resource-identifier = *(ALPHA / DIGIT / "-" / "." / "_" / "~" / "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "=" / "@")
+```
+
+_Note: if we'd like to use resource identifier in URL query the charset is significantly smaller:_
+
+```
+resource-identifier = *(ALPHA / DIGIT / "-" / "." / "_" / "~" )
+```
+
+#### What characters we can use as a delimiter?
+
+As a delimiter we can use only characters that can't be present in resource identifier components – `name`, `mesh`, `namespace`, `zone`, `resourceType` and `sectionName`:
+
+* `name` and `zone`
+  * contain no more than 253 characters
+  * contain only lowercase alphanumeric characters, '-' or '.'
+  * start with an alphanumeric character
+  * end with an alphanumeric character
+* `mesh`
+  * contain no more than 63 characters
+  * contain only lowercase alphanumeric characters, '-' or '.'
+  * start with an alphanumeric character
+  * end with an alphanumeric character
+* `namespace`
+  * contain at most 63 characters
+  * contain only lowercase alphanumeric characters or '-'
+  * start with an alphanumeric character
+  * end with an alphanumeric character
+* `resourceType`
+  * only alphanumeric
```

Comment:
There's no validation for this atm added: https://github.com/kumahq/kuma/issues/12861